File: CS/WebSite/App_Code/TestData.cs
using System;
using System.Collections.Generic;
using System.Web;
using System.Web.SessionState;
using System.ComponentModel;

public class DataRow {
    private int fId;
    private string fDescription;

	public DataRow(int id, string description){
        fId = id;
        fDescription = description;
    }

    public int ID {
        get { return fId; }
        set { fId = value; }
    }

    public string Description {
        get { return fDescription;}
        set { fDescription = value; } 
    }
}

public class GridDataSource {
    HttpSessionState Session { get { return HttpContext.Current.Session; } }

    public BindingList<DataRow> GetRows() {
        if (Session["GridDataSource"] == null) CreateRows();
        return Session["GridDataSource"] as BindingList<DataRow>;
    }

    public void CreateRows() {
        BindingList<DataRow> res = new BindingList<DataRow>();
        for (int i = 1; i <= 25; i++){
            DataRow item = new DataRow(i, "Sample data for row " + i);
            res.Add(item);
        }
        Session["GridDataSource"] = res;
    }
    
    public void DeleteRow(int id) {
        BindingList<DataRow> rows = Session["GridDataSource"] as BindingList<DataRow>;
        for (int i = 0; i < rows.Count; i++) {
            if (rows[i].ID == id){
                rows.Remove(rows[i]);
                break;
            }
        }
    }
}

File: VB/Solution/App_Code/TestData.vb
Imports System.ComponentModel

Public Class DataRow
    Private fId As Integer
    Private fDescription As String

    Public Sub New(ByVal id As Integer, ByVal description As String)
        fId = id
        fDescription = description
    End Sub

    Public Property ID() As Integer
        Get
            Return fId
        End Get
        Set(ByVal value As Integer)
            fId = value
        End Set
    End Property

    Public Property Description() As String
        Get
            Return fDescription
        End Get
        Set(ByVal value As String)
            fDescription = value
        End Set
    End Property
End Class

Public Class GridDataSource
    Private ReadOnly Property Session() As HttpSessionState
        Get
            Return HttpContext.Current.Session
        End Get
    End Property

    Public Function GetRows() As BindingList(Of DataRow)
        If Session("GridDataSource") Is Nothing Then
            CreateRows()
        End If
        Return TryCast(Session("GridDataSource"), BindingList(Of DataRow))
    End Function

    Public Sub CreateRows()
        Dim res As BindingList(Of DataRow) = New BindingList(Of DataRow)()
        For i As Integer = 1 To 25
            Dim item As New DataRow(i, "Sample data for row " & i)
            res.Add(item)
        Next i
        Session("GridDataSource") = res
    End Sub

    Public Sub DeleteRow(ByVal id As Integer)
        Dim rows As BindingList(Of DataRow) = TryCast(Session("GridDataSource"), BindingList(Of DataRow))
        Dim i As Integer = 0
        Do While i < rows.Count
            If rows(i).ID = id Then
                rows.Remove(rows(i))
                Exit Do
            End If
            i += 1
        Loop
    End Sub
End Class